samples/ironside_se/wdtstart_sample/src/main.c

```diff
@@ -0,0 +1,148 @@
+	}
+}
+
+static int init_watchdog(void)
+{
+	wdt_dev = DEVICE_DT_GET_OR_NULL(DT_NODELABEL(wdt011));
+	if (wdt_dev == NULL) {
+		printk("Watchdog device not found\n");
+		return -ENODEV;
+	}
+
+	if (!device_is_ready(wdt_dev)) {
+		printk("Watchdog device not ready\n");
+		return -ENODEV;
+	}
+
+	/* Configure timeout to match UICR.WDTSTART configuration */
+	struct wdt_timeout_cfg wdt_config = {
+		.window =
+			{
+				.min = 0, .max = 2000 /* 2 seconds in milliseconds */
+			},
+		.callback = NULL,
+		.flags = WDT_FLAG_RESET_SOC};
+
+	wdt_channel_id = wdt_install_timeout(wdt_dev, &wdt_config);
+	if (wdt_channel_id < 0) {
+		printk("Failed to install watchdog timeout: %d\n", wdt_channel_id);
+		return wdt_channel_id;
+	}
+
+	/* Setup watchdog with minimal options */
+	int ret = wdt_setup(wdt_dev, 0);
+
+	if (ret != 0) {
+		printk("Failed to setup watchdog: %d\n", ret);
+		return ret;
+	}
+
+	printk("Watchdog driver initialized successfully\n");
+	return 0;
+}
+
+static void feed_watchdog(void)
+{
+	int ret = wdt_feed(wdt_dev, wdt_channel_id);
+
+	if (ret != 0) {
+		printk("Failed to feed watchdog: %d\n", ret);
+	}
+}
+
+int main(void)
+{
+	int err;
+	uint32_t reset_cause = 0;
+
+	printk("=== IronSide SE WDTSTART Sample ===\n\n");
+
+	/* Check reset cause using hwinfo API */
+	check_reset_cause();
+
+	err = hwinfo_get_reset_cause(&reset_cause);
+	if (err == 0 && (reset_cause & RESET_WATCHDOG)) {
+		/* Watchdog reset detected - continuously feed the watchdog */
+		printk("\nWatchdog reset detected - continuously feeding watchdog\n");
+		printk("This demonstrates that we can detect and handle watchdog resets\n\n");
+
+		/* Initialize watchdog driver.
+		 *
+		 * Note that this is technically a reinitialization as the
+		 * watchdog HW has already been configured and started by IronSide
+		 * SE.
+		 *
+		 * The Zephyr driver gracefully handles intialization of an
+		 * already-initialized watchdog.
+		 */
+		err = init_watchdog();
+		if (err != 0) {
+			printk("Failed to initialize watchdog driver: %d\n", err);
+			return err;
+		}
+
+		/* Continuously feed the watchdog */
+		int feed_count = 0;
+
+		while (1) {
+			feed_watchdog();
+			feed_count++;
+
+			if (feed_count % 500 == 1) {
+				printk("Watchdog fed %d times - system still running!\n",
+				       feed_count);
+			}
+
+			k_sleep(K_MSEC(100));
+		}
+	} else {
+		/* No watchdog reset - wait for watchdog to trigger */
+		printk("\nNo watchdog reset detected - waiting for watchdog to trigger...\n");
+		printk("The watchdog should timeout and reset the system shortly.\n");
+		printk("After the reset, this sample will detect the watchdog reset\n");
+		printk("and continuously feed the watchdog to prevent further resets.\n\n");
+
+		/* Wait for watchdog to trigger - don't feed it */
+		printk("Waiting for watchdog timeout (not feeding watchdog)...\n");
+		while (1) {
+			printk("Still waiting for watchdog to trigger...\n");
+			k_sleep(K_MSEC(10000));
```

Comment:
nit: maybe we should switch the order of the wait and the print so that the first "Still waiting" message is printed after 1 sec instead of immediately